docs/types/reference-types.rst

```diff
@@ -27,8 +27,8 @@ Every reference type has an additional
 annotation, the "data location", about where it is stored. There are three data locations:
 ``memory``, ``storage`` and ``calldata``. Calldata is a non-modifiable,
 non-persistent area where function arguments are stored, and behaves mostly like memory.
-It is required for parameters of external functions but can also be used for other variables.
 
+Calldata is usually preferred over memory for external functions but is not required after version 0.6.9.
```

Comment:
I'd put the info about 0.6.9 in a note (just like the notes below).

You could also clarify that `calldata` is recommended because it's cheaper because it lets you skip ABI decoding of the parameter if it's not needed.

Reply:
I thought this was a pre-0.6.9 leftover but strangely, it looks like it was added in the PR that implemented the feature (#8942). I think that `It` might have originally meant the location annotation in general rather than `calldata` specifically. But I agree that it should be reworded. If that was the intention then it did not even occur to me without looking at the diff :)